tfx/examples/cifar10/cifar10_utils_native_keras.py

```diff
@@ -0,0 +1,398 @@
+  # Here we add label file to output metadata.
+  output_meta = _metadata_fb.TensorMetadataT()
+  label_file = _metadata_fb.AssociatedFileT()
+  label_file.name = os.path.basename(label_map_path)
+  label_file.type = _metadata_fb.AssociatedFileType.TENSOR_AXIS_LABELS
+  output_meta.associatedFiles = [label_file]
+
+  # Creates subgraph to contain input and output information,
+  # and add subgraph to the model information.
+  subgraph = _metadata_fb.SubGraphMetadataT()
+  subgraph.inputTensorMetadata = [input_meta]
+  subgraph.outputTensorMetadata = [output_meta]
+  model_meta.subgraphMetadata = [subgraph]
+
+  # Serialize the model metadata buffer we created above using flatbuffer builder.
+  b = flatbuffers.Builder(0)
+  b.Finish(
+      model_meta.Pack(b),
+      _metadata.MetadataPopulator.METADATA_FILE_IDENTIFIER)
+  metadata_buf = b.Output()
+
+  # Populates metadata and label file to the model file.
+  populator = _metadata.MetadataPopulator.with_model_file(model_path)
+  populator.load_metadata_buffer(metadata_buf)
+  populator.load_associated_files([label_map_path])
+  populator.populate()
+
+
+# TFX Trainer will call this function.
+def run_fn(fn_args: TrainerFnArgs):
+  """Train the model based on given args.
+
+  Args:
+    fn_args: Holds args used to train the model as name/value pairs.
+  """
+  tf_transform_output = tft.TFTransformOutput(fn_args.transform_output)
+
+  train_dataset = _input_fn(fn_args.train_files, tf_transform_output,
+                            is_train=True, batch_size=_TRAIN_BATCH_SIZE)
+  eval_dataset = _input_fn(fn_args.eval_files, tf_transform_output,
+                           is_train=False, batch_size=_EVAL_BATCH_SIZE)
+
+  mirrored_strategy = tf.distribute.MirroredStrategy()
+  with mirrored_strategy.scope():
+    model, base_model = _build_keras_model()
+
+  try:
+    log_dir = fn_args.model_run_dir
+  except KeyError:
+    # TODO(b/158106209): use ModelRun instead of Model artifact for logging.
+    log_dir = os.path.join(os.path.dirname(fn_args.serving_model_dir), 'logs')
+
+  absl.logging.info('Tensorboard logging to {}'.format(log_dir))
+  # Write logs to path
+  tensorboard_callback = tf.keras.callbacks.TensorBoard(
+      log_dir=log_dir, update_freq='batch')
+
+  # Our training regime has two phases: we first freeze the backbone and train
+  # the newly added classifier only, then unfreeze part of the backbone and
+  # fine-tune with classifier jointly.
+  steps_per_epoch = int(_TRAIN_DATA_SIZE / _TRAIN_BATCH_SIZE)
+  total_epochs = int(fn_args.train_steps / steps_per_epoch)
+  if _CLASSIFIER_EPOCHS > total_epochs:
+    raise Exception('Classifier epochs is greater than the total epochs')
+
+  absl.logging.info('Start training the top classifier')
+  # Freeze the whole MobileNet backbone and train the top classifer only
+  _freeze_model_by_percentage(base_model, 1.0)
+  # We need to recompile the model because layer properties have changed
+  model.compile(
```

Comment:
this compile should be within with mirrored_strategy.scope():? or just drop the strategy support

Reply:
Sounds good. I have moved model.compile under mirrored_strategy.

Reply:
Is this tested？I'm not sure if it can be use for two model training in sequence, if not tested, consider just drop the support

Reply:
It seems it works fine for running on cpus. I'll drop the support for now.